cinema/views.py

```diff
@@ -31,10 +38,39 @@ class CinemaHallViewSet(viewsets.ModelViewSet):
 
 
 class MovieViewSet(viewsets.ModelViewSet):
-    queryset = Movie.objects.all()
+    queryset = Movie.objects.prefetch_related("genres", "actors")
     serializer_class = MovieSerializer
 
-    def get_serializer_class(self):
+    @staticmethod
+    def _params_to_ints(qs: str) -> list:
+        """Converts a list of string IDs to a list of ints"""
+        return [int(str_id) for str_id in qs.split(",")]
+
+    def get_queryset(self) -> QuerySet:
+        queryset = self.queryset
+
+        actors = self.request.query_params.get("actors")
+        genres = self.request.query_params.get("genres")
+        title = self.request.query_params.get("title")
+
+        if actors:
+            actors_ids = self._params_to_ints(actors)
+            queryset = queryset.filter(actors__id__in=actors_ids)
+
+        if genres:
+            genres_ids = self._params_to_ints(genres)
+            queryset = queryset.filter(genres__id__in=genres_ids)
+
+        if title:
+            queryset = queryset.filter(title__icontains=title)
+
+        return queryset.distinct()
+
+    def get_serializer_class(self) -> Type[
+        MovieListSerializer
+        | MovieDetailSerializer
+        | MovieSerializer
+    ]:
```

Comment:
```suggestion
    def get_serializer_class(self) -> Type[MovieSerializer]:
```

Because MovieListSerializer and MovieDetailSerializer inherit from MovieSerializer